src/mjolnir/landmark_database_builder.cc

```diff
@@ -0,0 +1,189 @@
+  if (ret != SQLITE_OK) {
+    LOG_ERROR("SQL error: " + sql);
+    LOG_ERROR(std::string(sqlite3_errmsg(db)));
+    sqlite3_close(db);
+    return false;
+  }
+
+  sqlite3_reset(stmt);
+  sqlite3_clear_bindings(stmt);
+
+  if (name != "") {
+    sqlite3_bind_text(stmt, 1, name.c_str(), name.length(), SQLITE_STATIC);
+  } else {
+    sqlite3_bind_null(stmt, 1);
+  }
+
+  if (type != "") {
+    sqlite3_bind_text(stmt, 2, type.c_str(), type.length(), SQLITE_STATIC);
+  } else {
+    sqlite3_bind_null(stmt, 2);
+  }
+
+  sqlite3_bind_text(stmt, 3, longitude.c_str(), longitude.length(), SQLITE_STATIC);
+  sqlite3_bind_text(stmt, 4, latitude.c_str(), latitude.length(), SQLITE_STATIC);
+
+  /* performing INSERT INTO */
+  ret = sqlite3_step(stmt);
+  if (ret == SQLITE_DONE || ret == SQLITE_ROW) {
+    sqlite3_finalize(stmt);
+    LOG_INFO("Inserted landmark");
+    return true;
+  }
+
+  LOG_ERROR("sqlite3_step() error: " + std::string(sqlite3_errmsg(db)));
+  LOG_ERROR("sqlite3_step() Name: " + name);
+  LOG_ERROR("sqlite3_step() Type: " + type);
+  LOG_ERROR("sqlite3_step() longitude: " + longitude);
+  LOG_ERROR("sqlite3_step() Latitude: " + latitude);
+
+  sqlite3_finalize(stmt);
+  return false;
+}
+
+bool LandmarkDatabase::createSpatialIndex() {
+  sql = "SELECT CreateSpatialIndex('landmarks', 'geom')";
+  ret = sqlite3_exec(db, sql.c_str(), NULL, NULL, &err_msg);
+  if (ret != SQLITE_OK) {
+    LOG_ERROR("Error: " + std::string(err_msg));
+    sqlite3_free(err_msg);
+    sqlite3_close(db);
+    return false;
+  }
+  LOG_INFO("Created spatial index");
+  return true;
+}
+
+static int test_callback(void *data, int argc, char **argv, char **azColName){
+   fprintf(stderr, "%s: \n", (const char*)data);
+   
+   for(int i = 0; i<argc; i++){
+      printf("%s = %s\n", azColName[i], argv[i] ? argv[i] : "NULL");
+   }
+   
+   printf("\n");
+   return 0;
+}
+
+bool LandmarkDatabase::testSelectQuery() {
+  sql = "SELECT name, type FROM landmarks where name = 'Eiffel Tower'";
+
+  const char* data = "Callback function called";
+  ret = sqlite3_exec(db, sql.c_str(), test_callback, (void*)data, &err_msg);
+  
+  if( ret != SQLITE_OK ) {
+    fprintf(stderr, "SQL error: %s\n", err_msg);
+    sqlite3_free(err_msg);
+    return false;
+  } else {
+    fprintf(stdout, "Operation done successfully\n");
+  }
+  return true;
+}
+
+int boundingBoxQueryCallback(void* data, int argc, char** argv, char** /* columnNames */) {
+  std::vector<std::pair<std::string, std::string>>* results = 
+    static_cast<std::vector<std::pair<std::string, std::string>>*>(data);
+
+  results->push_back(std::make_pair(argv[0], argv[1]));
+
+  return 0;
+}
+
+bool LandmarkDatabase::getLandmarksInBoundingBox(std::vector<std::pair<std::string, std::string>> *landmarks, 
+      const std::string& minLat, const std::string& minLong, const std::string& maxLat, const std::string& maxLong) {
+  sql  = "SELECT name, type FROM landmarks WHERE ST_Covers(BuildMbr(" +
```

Comment:
this should also select and return the geom. you should use st_intersects to future proof this. to create the bounding box you need to use st_envelope and probably also st_setsrid to tell it that it's the lat lon projection

Reply:
ah i see maybe `buildmbr` just needs you to also pass the srid 4326?

Reply:
I tried to parse srid 4326 but didn't work. If not clear to you as well I can try to parse real wkt data and see if it works.

Reply:
it really shouldnt be wkt in the end. also the arguments should be doubles not strings. i cant play with it myself at the moment but if you make some of the changes suggested i can take a look maybe at night

Reply:
no problem. whenerver you are free. I'm working on the code reviews and will make some changes but probably not all of them